src/mod/endpoints/mod_sofia/sofia_glue.c

```diff
@@ -924,7 +924,11 @@ char *sofia_glue_get_extra_headers(switch_channel_t *channel, const char *prefix
 			if (!strncasecmp(name, prefix, strlen(prefix))) {
 				if ( !exclude_regex || !(proceed = switch_regex_perform(name, exclude_regex, &re, ovector, sizeof(ovector) / sizeof(ovector[0])))) {
 					const char *hname = name + strlen(prefix);
-					stream.write_function(&stream, "%s: %s\r\n", hname, value);
+					if (!strcasecmp(hname, "identity_div")) {
+						stream.write_function(&stream, "%.8s: %s\r\n", hname, value);
```

Comment:
Hmm, this is a workoround, since we already have support for IDENTITY header in sofia (my changes). This enables to set IDENTITY headers in FS using SIPTAG_IDENTITY_STR(identity) macro, e.g:
TAG_IF(!zstr(sih), SIPTAG_IDENTITY_STR(sih))
It's fine, but using SIPTAG_IDENTITY_STR is how identity header is supported by sofia.

Reply:
So something like TAG_IF(!zstr(identity), SIPTAG_IDENTITY_STR(identiy)), TAG_IF(!zstr(identity_div), SIPTAG_IDENTITY_STR(identiy_div)) around lines 1696 in sofia_glue.c (sofia_glue_do_invite) should work.

Reply:
we would have to change the channel var, instead of sip_h_identity, we would use sip_identity instead (sip_identity_div for diversion). Is this good?

Reply:
sip_identity and sip_identiy_div is fine to me, we can change them anytime if required anyway